cpp/oneapi/dal/detail/error_messages.cpp

```diff
@@ -164,6 +166,8 @@ MSG(unknown_distance_type,
 MSG(distance_is_not_supported_for_gpu, "Only Euclidean distances for k-NN is supported for GPU")
 MSG(incompatible_knn_model,
     "The provided model is incompatible with the selected k-NN task or method")
+MSG(invalid_set_of_result_options_to_search,
+    "Invalid set of optional results to search. Search task can not return responses.")
```

Comment:
There is no "optional results" in the lib, there are only "result options".

Suggestion: Provided results options are incompatible with the search task. Search task cannot compute responses.

Reply:
Applied